src/core/tsi/s2a/handshaker/s2a_tsi_handshaker.h

```diff
@@ -21,64 +21,78 @@
 
 #include <grpc/grpc.h>
 
+#include <string.h>
 #include "src/core/lib/iomgr/pollset_set.h"
 #include "src/core/tsi/s2a/handshaker/s2a_handshaker_client.h"
 #include "src/core/tsi/transport_security.h"
 #include "src/core/tsi/transport_security_interface.h"
 #include "src/proto/grpc/gcp/s2a.upb.h"
 
-namespace grpc {
+namespace grpc_core {
 namespace experimental {
 
 typedef struct s2a_tsi_handshaker s2a_tsi_handshaker;
 
 /** This method populates |self| with an instance of the s2a_tsi_handshaker,
  *  which is configured using the other arguments.
- * - options: S2A-specific options used to configure the s2a_tsi_handshaker.
- * - target_name: the name of the endpoint to which the channel connects; this
- *   data will be used for a secure naming check.
- * - handshaker_service_url: the address of S2A handshaker service; it follows
- *   the format "host:port".
- * - is_client: a boolean that is true if |client| is used at the client side,
- *   and false if |client| is used at the server side.
- * - interested_parties: set of pollsets interested in this connection.
- * - self: the address of S2A TSI handshaker instance to be populated by the
- *   method; the caller must ensure that |self| is not nullptr.
- * - error_details: an error message for when the creation fails. It is legal
- *   (and expected) to have |error_details| point to a nullptr.
+ *  - options: S2A-specific options used to configure the s2a_tsi_handshaker.
+ *  - target_name: the name of the endpoint to which the channel connects; this
+ *    data will be used for a secure naming check.
+ *  - is_client: a boolean that is true if |client| is used at the client side,
+ *    and false if |client| is used at the server side.
+ *  - interested_parties: set of pollsets interested in this connection.
+ *  - self: the address of S2A TSI handshaker instance to be populated by the
+ *    method; the caller must ensure that |self| is not nullptr.
+ *  - error_details: an error message for when the creation fails. It is legal
+ *    (and expected) to have |error_details| point to a nullptr.
  *
  * It returns TSI_OK on success and an error status code on failure. Note that
```

Comment:
Does it still hold for S2A?

Reply:
Are you referring to the `interested_parties` comment? If so, I should have removed this comment, since we are not expecting any special handling of the case `interested_parties == nullptr`, at least for the moment.

Reply:
Sorry for the confusion and Yes, the comment that is confusing is - "if interested_parties is nullptr...". Please remove it.